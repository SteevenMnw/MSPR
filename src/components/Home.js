//import liraries
import React from "react";
import { View, Text, StyleSheet } from "react-native";

// create a component
const Home = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Custom bottom tab nav !</Text>
    </View>
  );
};

// define your styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#76a6ef",
  },
  text: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#fff",
  },
});

//make this component available to the app
export default Home;
