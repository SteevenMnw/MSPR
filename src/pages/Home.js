//import liraries
import React, { useEffect } from "react";
import { View, Text, StyleSheet } from "react-native";
import { getAllCoupons } from "../API/API_Access";

import { getUserById } from "../API/API_Access";

import AsyncStorage from "@react-native-async-storage/async-storage";

// create a component
const Home = () => {
  useEffect(() => {
    getAllCoupons().then((res) => res);
    getUserById(1).then((data) => {
      setUserSession(data);
    });
  }, []);
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Custom bottom tab nav !</Text>
    </View>
  );
};

setUserSession = async (value) => {
  try {
    const jsonValue = JSON.stringify(value);
    await AsyncStorage.setItem("user", jsonValue);
  } catch (e) {
    console.log(e);
  }
};

// define your styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#76a6ef",
  },
  text: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#fff",
  },
});

//make this component available to the app
export default Home;
