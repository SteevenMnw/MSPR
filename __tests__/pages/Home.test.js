import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Home from "../../src/pages/Home";
//import {fireEvent, render} from '@testing-library/react-native'
import { expect, it } from "@jest/globals";

jest.useFakeTimers();

jest.mock("react-native-tiny-toast", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));

// Unit test (https://docs.expo.io/guides/testing-with-jest/#unit-test)
describe("<Home />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  // Snapshop de l'ensemble de la page Home
  it("renders correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
