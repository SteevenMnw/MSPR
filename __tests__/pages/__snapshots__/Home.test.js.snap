// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E8E8E8",
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
